src/App/Lobby/Room.tsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import getMessagePage from '../MessagePage';
+import { LobbyService, IRoomMetadata, IPlayerInRoom } from './LobbyService';
+import AlertLayer from '../Game/AlertLayer';
+import Card from '@material-ui/core/Card';
+import Button from '@material-ui/core/Button';
+import Typography from '@material-ui/core/Typography';
+import CardContent from '@material-ui/core/CardContent';
+import Tooltip from '@material-ui/core/Tooltip';
+import IconButton from '@material-ui/core/IconButton';
+import TextField from '@material-ui/core/TextField';
+
+interface IExpectedParams {
+  gameCode: string;
+  roomID: string;
+}
+interface IRoomProps {
+  match: { params: IExpectedParams };
+  history?: { push: (url: string) => void };
+}
+
+interface IRoomState {
+  room: IRoomMetadata;
+  player?: IPlayerInRoom;
+  nameTextBox: string;
+}
+
+export class Room extends React.Component<IRoomProps, IRoomState> {
+  private timer: any;  // fixme
+  constructor(props: IRoomProps) {
+    super(props);
+    const roomID = this.props.match.params.roomID;
+    this.state = {
+      nameTextBox: '',
+      room: {
+        gameCode: this.props.match.params.gameCode,
+        ready: false,
+        players: [],
+        roomID,
+      },
+      player: LobbyService.getCredential(roomID),
```

Comment:
this is inconsistent an confusing. The same data you called PlayerInRoom, Credential, and Player.